source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -700,16 +702,97 @@ def apply_final_journey_filters_post_finalize(response_list, request):
         journey_pairs_pool = itertools.combinations(journeys, 2)
         _filter_similar_line_and_crowfly_journeys(journey_pairs_pool, request)
 
-    keep_olympics_journeys(response_list, request)
 
+def get_journey_pt_extremity(journey, clockwise):
+    if clockwise:
+        sections = reversed(journey.sections)
+    else:
+        sections = journey.sections
+    extemity_pt_section = next((s for s in sections if s.type == response_pb2.PUBLIC_TRANSPORT), None)
+
+    assert extemity_pt_section
+    if clockwise:
+        return extemity_pt_section.destination
+    return extemity_pt_section.arrival
+
+
+class Interval:
+    def __init__(self, minimum, maximum):
+        self.minimum = minimum
+        self.maximum = maximum
 
-def keep_olympics_journeys(responses, request):
-    keep_olympics_journeys = request.get("_keep_olympics_journeys", False)
-    if keep_olympics_journeys:
-        for response in responses:
-            for journey in response.journeys:
-                if 'to_delete' in journey.tags and 'olympics' in journey.tags:
-                    journey.tags.remove("to_delete")
+    def includes(self, n):
+        return self.minimum <= n <= self.maximum
+
+
+def filter_olympics_journeys(responses, request):
+    if request.get('debug', False):
+        return
+
+    if not request.get("_keep_olympics_journeys", False):
+        return
+
+    intervals = {
+        # 9
+        "stop_point:IDFM:monomodalStopPlace:58498": Interval(0, 4240),
+        # 6
+        "stop_point:IDFM:22041": Interval(4240, 5606),
+        "stop_point:IDFM:463281": Interval(4240, 5606),
+        # 4
+        "stop_point:IDFM:412988": Interval(5606, float('inf')),
+        "stop_point:IDFM:412987": Interval(5606, float('inf')),
+    }
+
+    attractivity = {
+        "stop_point:IDFM:monomodalStopPlace:58498": 9,
+        "stop_point:IDFM:22041": 6,
+        "stop_point:IDFM:463281": 6,
+        "stop_point:IDFM:412988": 4,
+        "stop_point:IDFM:412987": 4,
+    }
+
+    olympics_durations = [j.duration for r in responses for j in r.journeys if 'olympics' in j.tags]
+    average_duration = sum(olympics_durations) / float(len(olympics_durations))
+
+    retained_journey = None
+
+    found_sps = []
+    for r in responses:
+        for j in r.journeys:
+            if 'olympics' not in j.tags:
+                continue
+            pt_extremity = get_journey_pt_extremity(j, request.get('clockwise'))
```

Comment:
I am not sure the `clockwise` parameter is the best one to decide if we should look at the first or last stop_point of the journey. 
The request could be "clockwise" but ends at a Olympics POI, or not "clockwise" and starts at an olympic POI. 

Maybe use the `criteria` ? 
- If `criteria == arrival_stop_attractivity` you should look at the last stop_point, 
- If `criteria == departure_stop_attractivity` you should look at the first stop_point,